tests/metrics-port/00-assert.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: metrics-port
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: metrics-port
+spec:
+  selector:
+    matchLabels:
+      app: metrics-port
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+      labels:
+        app: metrics-port
+    spec:
+      containers:
+        - name: metrics-port
+          image: image
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+              protocol: TCP
+            - containerPort: 8181
+              protocol: TCP
+          securityContext:
+            allowPrivilegeEscalation: false
+            privileged: false
+            readOnlyRootFilesystem: true
+            runAsGroup: 150
+            runAsUser: 150
+            seccompProfile:
+              type: RuntimeDefault
+          volumeMounts:
+            - mountPath: /tmp
+              name: tmp
+      imagePullSecrets:
+        - name: github-auth
+      securityContext:
+        fsGroup: 150
+        supplementalGroups:
+          - 150
+      serviceAccountName: metrics-port
+      volumes:
+        - emptyDir: {}
+          name: tmp
+---
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: metrics-port
+spec:
+  minReplicas: 1
+  maxReplicas: 1
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: 80
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: metrics-port
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: metrics-port
+spec:
+  selector:
+    app: metrics-port
+  ports:
+    - name: http
+      port: 8080
+      targetPort: 8080
+      protocol: TCP
+      appProtocol: http
```

Comment:
A couple of thoughts:
- Why is the port named `http` and `appProtocol` (not required) defined? We really don't know if the traffic is HTTP only.
- We should consider if we should expose other ports here as well (generic or only `metrics`). If going the Prometheus route later on, it is a requirement for the `ServiceMonitor` CRD.

Reply:
How Skiperator is set up to handle these things I don't know, I will only note that these points are out of scope in regards to the original Jira task (as we explicity do not create any services or Istio CRs because Instana does not need these in order to reach the pod), and the test assert in question is mostly copy-pasted from another test.

Reply:
We could push the non-pod aspects for a later issue 👍🏻